matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/content/ThumbnailExtractor.kt

```diff
@@ -33,34 +35,50 @@ internal object ThumbnailExtractor {
             val mimeType: String
     )
 
-    fun extractThumbnail(attachment: ContentAttachmentData): ThumbnailData? {
-        val file = File(attachment.path)
-        if (!file.exists() || !file.isFile) {
-            return null
-        }
+    fun extractThumbnail(context: Context, attachment: ContentAttachmentData): ThumbnailData? {
         return if (attachment.type == ContentAttachmentData.Type.VIDEO) {
-            extractVideoThumbnail(attachment)
+            extractVideoThumbnail(context, attachment)
         } else {
             null
         }
     }
 
-    private fun extractVideoThumbnail(attachment: ContentAttachmentData): ThumbnailData? {
-        val thumbnail = ThumbnailUtils.createVideoThumbnail(attachment.path, MediaStore.Video.Thumbnails.MINI_KIND) ?: return null
-        val outputStream = ByteArrayOutputStream()
-        thumbnail.compress(Bitmap.CompressFormat.JPEG, 100, outputStream)
-        val thumbnailWidth = thumbnail.width
-        val thumbnailHeight = thumbnail.height
-        val thumbnailSize = outputStream.size()
-        val thumbnailData = ThumbnailData(
-                width = thumbnailWidth,
-                height = thumbnailHeight,
-                size = thumbnailSize.toLong(),
-                bytes = outputStream.toByteArray(),
-                mimeType = "image/jpeg"
-        )
-        thumbnail.recycle()
-        outputStream.reset()
+    private fun extractVideoThumbnail(context: Context, attachment: ContentAttachmentData): ThumbnailData? {
+        var thumbnailData: ThumbnailData? = null
+        val mediaMetadataRetriever = MediaMetadataRetriever()
+        try {
+            mediaMetadataRetriever.setDataSource(context, attachment.queryUri)
+            var thumbnail = mediaMetadataRetriever.frameAtTime
```

Comment:
not sure if it's the best frame choice. when capturing on mobile it's often blurry? 
I think createVideoThumbnail is a bit smartter

Reply:
Couldn't use createVideoThumbnail since it requires the original path. I removed resizing logic for better resolution.